libs/NfcCoreLib/lib/Nci/phNciNfc.c

```diff
@@ -677,6 +677,70 @@ NFCSTATUS phNciNfc_Nfcee_ModeSet(void * pNciHandle,
     return wStatus;
 }
 
+NFCSTATUS phNciNfc_Nfcee_SePowerAndLinkCtrlSet(void * pNciHandle,
+                                               void * pNfceeHandle,
+                                               phNciNfc_PowerLinkModes_t eActivationMode,
+                                               pphNciNfc_IfNotificationCb_t pNotifyCb,
+                                               void *pContext)
+{
+    NFCSTATUS wStatus = NFCSTATUS_SUCCESS;
+    phNciNfc_Context_t * pNciContext = (phNciNfc_Context_t *)pNciHandle;
+    pphNciNfc_NfceeDeviceHandle_t pDevHandle = \
+        (pphNciNfc_NfceeDeviceHandle_t)pNfceeHandle;
+    uint8_t *pTargetInfo;
+
+    PH_LOG_NCI_FUNC_ENTRY();
+    if (NULL == pNciContext)
+    {
+        PH_LOG_NCI_CRIT_STR("Stack not initialized (phNciNfc_Nfcee_SePowerAndLinkCtrlSet)");
+        wStatus = NFCSTATUS_NOT_INITIALISED;
+    }
+    else if (NULL == pNotifyCb)
+    {
+        PH_LOG_NCI_CRIT_STR("Invalid parameter passed(phNciNfc_Nfcee_SePowerAndLinkCtrlSet)");
+        wStatus = NFCSTATUS_INVALID_PARAMETER;
+    }
+    else
+    {
+        if ((NULL != pDevHandle) && \
+            (0 != pDevHandle->tDevInfo.bNfceeID) && \
+            (PHNCINFC_INVALID_DISCID != pDevHandle->tDevInfo.bNfceeID))
+        {
+            pTargetInfo = (uint8_t *)phOsalNfc_GetMemory(PHNCINFC_NFCEEMODESET_PAYLOADLEN);
+            if (NULL == pTargetInfo)
+            {
+                PH_LOG_NCI_CRIT_STR("Memory not available(phNciNfc_Nfcee_SePowerAndLinkCtrlSet)");
+                wStatus = NFCSTATUS_INSUFFICIENT_RESOURCES;
+            }
+            else if (phNciNfc_IsVersion2x(pNciContext))
```

Comment:
Shouldn't you check the NCI version before allocation memory for the payload?

Reply:
That's true. I am adding a fix.